react-app/src/components/property/PropertyForms.tsx

```diff
@@ -454,35 +454,92 @@ interface INetBookValue {
 
 // Property.Fiscals
 export const NetBookValue = (props: INetBookValue) => {
+  const { years } = props;
+  // Sort the years array in descending order
+  const sortedYears = years.sort((a, b) => b - a);
+  const currentYear = sortedYears.at(0);
+  const { getValues } = useFormContext();
+
+  const handletheYearChange = () => {
+    const currentValues = getValues();
+    // Collect all the year values from the form
+    const yearValues = currentValues.Fiscals.map((fiscal: { FiscalYear: string }) =>
+      parseInt(fiscal.FiscalYear),
+    );
+    // Check for duplicates in the yearValues array
+    const yearCounts = yearValues.reduce((acc, year) => {
+      acc[year] = (acc[year] || 0) + 1;
+      return acc;
+    }, {});
+
+    for (const year in yearCounts) {
+      if (yearCounts[year] > 1) {
+        return `There is already a net book value for the year ${year}`;
+      }
+    }
```

Comment:
I can't seem to get this to iterate. I don't think objects are iterable in Javascript. You would need something like Object.entries(yearCounts) to get an iterable array.